tools/mksurfdata_map/src/mksurfdat.F90

```diff
@@ -1507,29 +1466,29 @@ subroutine normalizencheck_landuse(ldomain)
        end if
        
        ! Roundoff error fix
-       suma = pctlak(n) + pctwet(n) + pcturb(n) + pctgla(n)
-       suma2 = pctnatpft(n)%get_pct_l2g() + pctcft(n)%get_pct_l2g()
+       suma = pctlak(n) + pctwet(n) + pcturb(n) + pctgla(n) + pctcft(n)%get_pct_l2g()
        if ( (suma < 100._r8 .and. suma > (100._r8 - 1.e-6_r8)) .or. &
-            (suma2 > 0.0_r8 .and. suma2 <  1.e-6_r8) ) then
-          write (6,*) 'Special land units near 100%, but not quite for n,suma =',n,suma
-          write (6,*) 'Adjusting special land units to 100%'
-          if (pctlak(n) >= 25._r8) then
-             pctlak(n) = 100._r8 - (pctwet(n) + pcturb(n) + pctgla(n))
-          else if (pctwet(n) >= 25._r8) then
-             pctwet(n) = 100._r8 - (pctlak(n) + pcturb(n) + pctgla(n))
-          else if (pcturb(n) >= 25._r8) then
-             pcturb(n) = 100._r8 - (pctlak(n) + pctwet(n) + pctgla(n))
-          else if (pctgla(n) >= 25._r8) then
-             pctgla(n) = 100._r8 - (pctlak(n) + pctwet(n) + pcturb(n))
+            (pctnatpft(n)%get_pct_l2g() > 0.0_r8 .and. pctnatpft(n)%get_pct_l2g() <  1.e-6_r8) ) then
+          write (6,*) 'Special plus crop land units near 100%, but not quite for n,suma =',n,suma
+          write (6,*) 'Adjusting special plus crop land units to 100%'
+          if (pctlak(n) >= 1.0_r8) then
```

Comment:
- [ ] Thanks for making the changes to this error check. What would you think about adding a comment with some notes explaining why we're doing this. Something like the following (or whatever makes the most sense to you):

```Fortran
! This roundoff error fix is needed to handle the situation where new_total_natveg_pct
! ends up near 0 but not exactly 0 due to roundoff issues. In this situation, we set the
! natveg landunit area to exactly 0 and put the remainder into some other landunit. Since
! the remainder is very small, it doesn't really matter which other landunit we add it to,
! so we just pick some landunit that already has at least 1% cover.
```

Reply:
Good idea, thanks. Done.